### ISSUE TITLE
|mass segmentation fault

### ISSUE THREAD

**Issue by ndeet** [urbit/urbit#2141]
Maybe related to #835 but not sure.

On Debian 10, urbit binary v0.10.1:

```
~xxxxxx-xxxxxx:dojo> |mass/
%userspace:
  %hoon: MB/4.545.064
  %zuse: MB/5.208.496
  %caches:
    %vane-a: MB/3.180.272
    %vane-b: KB/290.028
    %vane-c: MB/5.802.328
    %vane-d: KB/895.924
    %vane-e: MB/1.968.816
    %vane-f: MB/5.700.280
    %vane-g: MB/1.847.900
    %vane-i: KB/277.080
    %vane-j: MB/1.215.424
  --MB/21.178.052
  %vanes:
    %dill:
      %hey: B/120
      %dug: KB/1.632
      %dot: B/384
    --KB/2.136
    %ames:
      %peers-known: B/0
      %peers-alien: KB/17.564
      %dot: B/980
    --KB/18.544
    %behn:
      %timers: KB/2.872
      %dot: B/248
    --KB/3.120
    %clay:
      %domestic: MB/7.377.540
      %foreign: KB/1.152
      %object-store:
        %commits: KB/93.696
        %blobs: KB/63.072
      --KB/156.768
      %dot: B/288
    --MB/7.535.748
    %eyre:
      %bindings: KB/4.196
      %auth: B/144
      %connections: KB/9.836
      %channels: KB/6.268
      %axle: B/448
    --KB/20.892
    %ford:
      %builds: MB/30.768.076
      %compiler-cache: MB/6.634.368
      %dot: KB/210.240
    --MB/37.612.684
    %gall:
      %foreign: B/192
      %blocked:
      --B/0
      %active:
        %acme: KB/1.296
        %azimuth-tracker: KB/1.880
        %chat-cli: MB/2.162.764
        %chat-hook: KB/1.944
        %chat-store: KB/998.288
        %chat-view: KB/4.144
        %clock: KB/2.248
        %dojo: KB/2.184
        %eth-watcher: MB/4.262.876
        %goad: KB/1.176
        %group-hook: KB/1.080
        %group-store: KB/2.808
        %hood: KB/16.224
        %invite-hook: KB/1.248
        %invite-store: KB/2.232
        %invite-view: KB/2.940
        %launch: KB/5.776
        %lens: KB/1.032
        %permission-group-hook: KB/1.248
        %permission-hook: KB/1.716
        %permission-store: KB/3.360
        %ping: KB/1.608
        %publish: KB/2.332
        %soto: KB/2.164
        %spider: KB/5.152
        %weather: KB/2.200
      --MB/7.491.920
      %dot: KB/29.764
    --MB/7.521.876
    %iris:
      %nex: B/0
      %outbound: B/160
      %by-id: B/0
      %by-duct: B/0
      %axle: B/96
    --B/256
    %jael:
      %pki: KB/985.448
      %etn: B/288
      %dot: B/48
    --KB/985.784
  --MB/53.701.040
  %vases:
    %vane-a: B/384
    %vane-b: B/312
    %vane-c: B/360
    %vane-d: B/408
    %vane-e: B/336
    %vane-f: B/336
    %vane-g: KB/2.004
    %vane-i: B/336
    %vane-j: B/336
  --KB/4.812
  %dot: KB/125.492
--MB/84.762.956
total userspace: MB/84.762.956
  kernel: KB/14.440
total arvo stuff: KB/14.440
  warm jet state: KB/80.008
  cold jet state: KB/57.948
  hank cache: B/372
  battery hash cache: B/288
  call site cache: B/100
  hot jet state: KB/128.048
total jet stuff: KB/266.764
  bytecode programs: MB/1.845.644
  bytecode cache: KB/166.652
total nock stuff: MB/2.012.296
  trace buffer: B/36
  memoization cache: B/288
total road stuff: B/324
space profile: KB/2.232
event: B/144
effects: KB/2.976
total marked: MB/87.062.132
free lists: MB/50.507.224
sweep: MB/87.062.132
> |mass
>=
measuring daemon:
  kernel: MB/7.213.972
  date: B/36
  formatted date: B/48
  instance string: B/32
  wish cache: B/856
total arvo stuff: MB/7.214.944
  warm jet state: KB/34.536
  cold jet state: KB/22.884
  hank cache: B/288
  battery hash cache: B/288
  hot jet state: KB/125.868
total jet stuff: KB/183.864
  bytecode programs: KB/892.624
  bytecode cache: KB/56.716
total nock stuff: KB/949.340
  trace buffer: B/36
  memoization cache: KB/8.756
total road stuff: KB/8.792
total marked: MB/8.356.940
leak: 0x36a7d1e0 mug=0 swept=1
    size: B/28
    data: (null)
leaked: B/28
Assertion '(0 == leq_w) && (0 == weq_w)' failed in noun/allocate.c:1992


bail: oops
bailing out
Assertion failed: 0 (noun/manage.c: u3m_bail: 667)
Segmentation fault (core dumped)
```

**Comment by ohAitch** [comment]
It's definitely the same "we leaked some memory" assert failing, possibly
different memory than in 2017 however~

**Comment by joemfb** [comment]
This is a leak in the daemon, not the worker, so it doesn't affect persistent memory. We should just stop checking the daemon on `|mass`, it's only useful in development.

**Comment by syzygyzer** [comment]
Bumping this since there's been a number of major updates over the past few months and it still exists. I get the same error as the OP. I am running NixOS 19.09 using the urxvt terminal emulator running bash.

```~tidrel-tabpub:dojo> |mass|
%userspace: 
  %hoon: MB/8.250.560
  %zuse: MB/5.336.996
  %caches: 
    %vane-a: MB/3.380.768
    %vane-b: KB/300.088
    %vane-c: MB/5.732.100
    %vane-d: KB/964.672
    %vane-e: MB/2.046.340
    %vane-f: MB/5.728.676
    %vane-g: MB/2.025.776
    %vane-i: KB/279.796
    %vane-j: MB/1.290.396
  --MB/21.748.612
  %vanes: 
    %dill: 
      %hey: B/120
      %dug: KB/1.632
      %dot: B/384
    --KB/2.136
    %ames: 
      %peers-known: B/0
      %peers-alien: MB/5.907.288
      %dot: MB/8.922.972
    --MB/14.830.260
    %behn: 
      %timers: KB/220.848
      %dot: B/248
    --KB/221.096
    %clay: 
      %domestic: MB/94.083.928
      %foreign: MB/62.453.384
      %object-store: 
        %commits: MB/12.274.972
        %blobs: MB/53.258.888
      --MB/65.533.860
      %dot: B/572
    --MB/222.071.744
    %eyre: 
      %bindings: KB/5.412
      %auth: B/576
      %connections: KB/13.176
      %channels: KB/20.312
      %axle: B/484
    --KB/39.960
    %ford: 
      %builds: MB/85.712.076
      %compiler-cache: MB/1.172.528
      %dot: KB/580.040
    --MB/87.464.644
    %gall: 
      %foreign: KB/8.544
      %blocked: 
      --B/0
      %active: 
        %acme: KB/2.664
        %azimuth-tracker: KB/2.016
        %chat-cli: MB/6.209.256
        %chat-hook: KB/22.424
        %chat-store: MB/23.256.816
        %chat-view: KB/24.436
        %clock: KB/2.328
        %contact-hook: KB/2.644
        %contact-store: KB/8.464
        %contact-view: KB/56.956
        %dojo: KB/11.628
        %eth-watcher: MB/4.526.420
        %goad: KB/1.284
        %group-hook: KB/2.220
        %group-store: KB/52.876
        %hood: KB/38.492
        %invite-hook: KB/1.476
        %invite-store: KB/5.180
        %invite-view: KB/83.116
        %launch: KB/89.512
        %lens: KB/1.164
        %link-listen-hook: KB/611.304
        %link-proxy-hook: KB/778.492
        %link-store: KB/57.080
        %link-view: KB/9.132
        %metadata-hook: KB/2.196
        %metadata-store: KB/93.664
        %permission-group-hook: KB/2.268
        %permission-hook: KB/2.484
        %permission-store: KB/33.952
        %ping: KB/1.656
        %publish: KB/65.816
        %soto: KB/2.172
        %spider: KB/2.128
        %weather: KB/2.588
      --MB/36.066.304
      %dot: MB/2.093.208
    --MB/38.168.056
    %iris: 
      %nex: B/0
      %outbound: B/160
      %by-id: B/0
      %by-duct: B/0
      %axle: B/132
    --B/292
    %jael: 
      %pki: MB/1.429.668
      %etn: B/288
      %dot: B/48
    --MB/1.430.004
  --MB/364.228.192
  %vases: 
    %vane-a: B/408
    %vane-b: B/192
    %vane-c: B/240
    %vane-d: B/288
    %vane-e: B/216
    %vane-f: B/216
    %vane-g: KB/1.908
    %vane-i: B/216
    %vane-j: B/216
  --KB/3.900
  %dot: KB/146.928
--MB/399.715.188
total userspace: MB/399.715.188
  kernel: KB/14.464
total arvo stuff: KB/14.464
  warm jet state: KB/94.732
  cold jet state: MB/124.186.384
  hank cache: B/372
  battery hash cache: B/288
  call site cache: B/100
  hot jet state: KB/127.840
total jet stuff: MB/124.409.716
  bytecode programs: MB/1.637.712
  bytecode cache: KB/148.308
total nock stuff: MB/1.786.020
  trace buffer: B/36
  memoization cache: B/288
total road stuff: B/324
space profile: KB/2.232
event: B/144
effects: KB/2.976
total marked: MB/525.931.064
free lists: MB/86.869.672
sweep: MB/525.931.064
> |mass
>=
measuring daemon:
  kernel: MB/7.229.516
  date: B/36
  formatted date: B/48
  instance string: B/32
  wish cache: B/856
total arvo stuff: MB/7.230.488
  warm jet state: KB/34.536
  cold jet state: KB/22.884
  hank cache: B/288
  battery hash cache: B/288
  hot jet state: KB/125.868
total jet stuff: KB/183.864
  bytecode programs: KB/894.848
  bytecode cache: KB/56.600
total nock stuff: KB/951.448
  trace buffer: B/36
  memoization cache: KB/8.688
total road stuff: KB/8.724
total marked: MB/8.374.524
leak: 0x200032d08 mug=0 swept=1
    size: B/28
    data: (null)
leaked: B/28
Assertion '(0 == leq_w) && (0 == weq_w)' failed in noun/allocate.c:1992

bail: oops
bailing out
Assertion failed: 0 (noun/manage.c: u3m_bail: 667)
Segmentation fault (core dumped)
```

I want to point out that the trailing | in "|mass|" is not something I typed, and an erroneous trailing character is also present in the OP. Running the command several times, sometimes it is a "/" instead of a "|", and sometimes there is no trailing character. This might be some minor Dill glitch that I'm guessing is unrelated to the issue at hand, but it seemed prudent to point it out just in case.

**Comment by jfranklin9000** [comment]
the `|` and `/` could be spinner residue

**Comment by joemfb** [comment]
That's right, they're spinner artifacts. `|mass` is printed to stderr (it can't really allocate due to subtleties of how it's implemented), so it's not properly interpolated into the console output.

In any case, this crash has been fixed as of urbit v0.10.4.
